### PR TITLE
Make ingress gateway HTTPS by default

### PR DESCRIPTION
**Pull request by DavidSpek**
**Which issue is resolved by this Pull Request:**
Closes: https://github.com/kubeflow/manifests/issues/1812 https://github.com/kubeflow/kubeflow/issues/5803

**Description of your changes:**
Currently the Jupyter, Tensorboards and Volumes Web Apps are setup to require secure cookies. However, as the istio ingress-gateway that Kubeflow uses is setup for HTTP, by default users will run into errors causing these web apps to not work. To mitigate this (and promote the use of HTTPS), this PR sets the ingress-gateway to use HTTPS by default with a certificate created by the `kubeflow-self-signing-issuer` that other kubeflow components already use.

/cc @yanniszark 

**Checklist:**
- [x] Unit tests pass:
  **Make sure you have installed kustomize == 3.2.1**
    1. `make generate-changed-only`
    2. `make test`

### CODE REVIEW

**Review by moficodes — APPROVED** [review]
/lgtm

**Comment by DavidSpek** [comment]
@moficodes You're missing a `/`, it's `/lgtm` :).

**Comment by moficodes** [comment]
I doubt my `/lgtm` does anything. But fixed it. 😄
Thanks once again.

**Comment by thesuperzapper** [comment]
@yanniszark we need you to approve (if you agree)

**Comment by yanniszark** [comment]
@thesuperzapper I am not convinced that this change is something we want to do. We can continue the discussion in https://github.com/kubeflow/manifests/issues/1812 and open a PR (or not) when we reach a conclusion. I believe @moficodes is unblocked to continue the integration with IBM, so I will come back to this after making a pass at other blocking issues.

**Comment by DavidSpek** [comment]
@yanniszark [Here](https://github.com/kubeflow/kubeflow/issues/5803#issuecomment-817310731) is another example of a user that is struggling with this problem even after following the instructions to setup HTTPS. If this was already part of the kustomize manifests he would not have run into this problem.

**Comment by yanniszark** [comment]
> @yanniszark Here is another example of a user that is struggling with this problem even after following the instructions to setup HTTPS. If this was already part of the kustomize manifests he would not have run into this problem.

Agreed, users will try to export Kubeflow over HTTP with NodePort / LoadBalancer / Ingress, that much is evident. For these reasons, I proposed: https://github.com/kubeflow/manifests/issues/1812#issuecomment-818117944

Let's take the discussion into the issue if you agree.

**Pull request closed without merging**
